braincache plugin

XSPGMike opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/XSPGMike/braincache_obsidian

## Release Checklist
- [ ] I have tested the plugin on
  - [ ]  Windows
  - [ ]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

liamcain commented:
Hi! Looks good, I mostly just made some comments about code style.

https://github.com/XSPGMike/braincache_obsidian/blob/36d3d6a307eb47b45f6dc6cb7939d53bab1e1a2a/src/main.ts#L47 style nitpick: I recommend giving this variable a lowercase name so it doesn't get mistaken for a class

https://github.com/XSPGMike/braincache_obsidian/blob/36d3d6a307eb47b45f6dc6cb7939d53bab1e1a2a/src/api.ts#L186 you probably want to add some error handling here so you're just just silently failing here

https://github.com/XSPGMike/braincache_obsidian/blob/36d3d6a307eb47b45f6dc6cb7939d53bab1e1a2a/src/api.ts#L187 Where are you handling the `undefined` values here? I don't see anything in applyPatches that expects to see `undefined`

https://github.com/XSPGMike/braincache_obsidian/blob/36d3d6a307eb47b45f6dc6cb7939d53bab1e1a2a/src/process.ts#L75 You might want a more descriptive name than "processDecks" here. Ideally it should be easier to tell what's happening inside the syncDecks plugin without reading the source of each of the helper functions

https://github.com/XSPGMike/braincache_obsidian/blob/36d3d6a307eb47b45f6dc6cb7939d53bab1e1a2a/src/main.ts#L58 cardCount here should be calculated based off `patches` since you're making separate API calls there, and it would be useful to report the number of successfully patched cards. Or even better `Synced ${patches.length} of ${totalCardCount} cards`

https://github.com/XSPGMike/braincache_obsidian/blob/36d3d6a307eb47b45f6dc6cb7939d53bab1e1a2a/src/settings.ts#L29 You should probably also add a login button to launch the login modal from settings

https://github.com/XSPGMike/braincache_obsidian/blob/36d3d6a307eb47b45f6dc6cb7939d53bab1e1a2a/src/commands.ts#L10-L15 You can refactor this to be:

```ts
const hasDeck = doc.includes('#deck');
editor.replaceRange(cardTemplate(hasDeck), editor.getCursor())
editor.setCursor(editor.getCursor().line - 3, 0)
```

https://github.com/XSPGMike/braincache_obsidian/blob/36d3d6a307eb47b45f6dc6cb7939d53bab1e1a2a/src/commands.ts#L6 super nitpicky but you might want to capitalize the command names to be consistent with the built-in commands

XSPGMike commented:
Hi, thank you for the in depth review.
I think I have addressed most of the issues you brought up, I pushed the patches to my repo.
[https://github.com/XSPGMike/braincache_obsidian](url)

XSPGMike commented:
hi, thanks for the in depth review.
I addressed most of your points and released a v0.0.2 in the plugin's repository.

liamcain commented:
Hi, sorry for taking a while to take another look!

https://github.com/XSPGMike/braincache_obsidian/blob/77171fcaba8abbe227ed1f482a10c8a0d08eec64/src/api.ts#L195 This line seems to assume that any response without a 200 or 201 code is still a successful API response. However, you might want to add special handling for 40X response or 50X responses from the API in case there was an issue with the request.

---

I don't think it's a blocker to go in though.

XSPGMike commented:
I see, yes I'll do that and then get back to you, thanks for the input.

XSPGMike commented:
Hi, I managed to optimize what you pointed out by making a single call to `vault.getMarkdownFiles()` and then immediately filtering content without splitting it by newline in `applyPatches`

XSPGMike commented:
ok, done, the release is out, thanks.
I'll read tags from the metadataCache in the next release then, I didn't quite get that.
